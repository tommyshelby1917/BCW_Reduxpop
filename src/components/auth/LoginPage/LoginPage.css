.loginPage {
  background-color: var(--yellowpop);
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
  justify-content: center;
}

.loginpage-main-container {
  background-color: var(--bluepop);
  border-radius: 50%;
  padding: 100px;
  color: #fff;
}

.loginForm-field {
  margin: 5px;
}

.loginForm-submit {
  background-color: #fff !important;
  color: var(--bluepop) !important;
}

.loginForm-submit:hover {
  transform: scale(0.8);
  transition: all 0.3s;
}
