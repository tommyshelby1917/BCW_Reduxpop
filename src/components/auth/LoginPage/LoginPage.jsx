import T from 'prop-types';
import { useState } from 'react';

import FormField from '../../common/FormField/FormField';
import Button from '../../common/Button/Button';

import { connect } from 'react-redux';
import { authLogin, iuResetError } from '../../../store/actions';
import { getUi } from '../../../store/selectors';

import './LoginPage.css';

export function LoginPage({ onLogin, deleteError, error }) {
  const [value, setValue] = useState({
    email: '',
    password: '',
    remember: false,
  });

  const handleChange = (event) => {
    setValue((prevState) => ({
      ...prevState,
      [event.target.name]:
        event.target.type === 'checkbox'
          ? event.target.checked
          : event.target.value,
    }));
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    deleteError();
    try {
      await onLogin(value);
    } catch (err) {
      console.log(err);
    }
  };

  return (
    <div className="loginPage">
      <div className="loginpage-main-container">
        <h1 className="loginPage-title">Log in to ReactPop!</h1>
        <form className="loginForm" onSubmit={handleSubmit}>
          <FormField
            type="email"
            name="email"
            label="Email: "
            className="loginForm-field"
            value={value.email}
            onChange={handleChange}
            autofocus
          ></FormField>
          <FormField
            type="password"
            name="password"
            label="Password: "
            className="loginForm-field"
            value={value.password}
            onChange={handleChange}
          ></FormField>
          <FormField
            type="checkbox"
            name="remember"
            label="Remember me"
            className="loginForm-field"
            value={value.remember}
            onChange={handleChange}
          ></FormField>
          <Button className="loginForm-submit" type="submit">
            Log in!!
          </Button>
        </form>
        {error && (
          <div onClick={deleteError} className="loginPage-error">
            {error.message}
          </div>
        )}
      </div>
    </div>
  );
}

LoginPage.propTypes = {
  onLogin: T.func.isRequired,
};

const mapStateToProps = (state) => {
  return getUi(state);
};

const mapDispatchToProps = {
  onLogin: authLogin,
  deleteError: iuResetError,
};

const ConnectedLoginPage = connect(
  mapStateToProps,
  mapDispatchToProps
)(LoginPage);

export default ConnectedLoginPage;
