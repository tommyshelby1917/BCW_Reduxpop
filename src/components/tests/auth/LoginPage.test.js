import { fireEvent, render } from '@testing-library/react';
import { LoginPage } from '../../auth/LoginPage/LoginPage';

describe('LoginPage', () => {
  test('snapshot', () => {
    const { container } = render(<LoginPage onLogin={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test('should call onLogin', () => {
    const onLogin = jest.fn().mockResolvedValue();
    const email = 'josep@josep.com';
    const password = '';
    const remember = false;

    const { getByLabelText, getByRole } = render(
      <LoginPage onLogin={onLogin} />
    );

    const emailField = getByLabelText(/Email/);
    const passwordField = getByLabelText(/Password/);
    const submitButton = getByRole('button');

    fireEvent.change(emailField, { target: { value: email } });
    fireEvent.change(passwordField, { target: { value: password } });

    fireEvent.click(submitButton);
    expect(onLogin).toHaveBeenCalledWith({ email, password, remember });
  });
});
