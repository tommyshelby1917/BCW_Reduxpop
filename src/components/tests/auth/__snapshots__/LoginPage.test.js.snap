// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<div>
  <div
    class="loginPage"
  >
    <div
      class="loginpage-main-container"
    >
      <h1
        class="loginPage-title"
      >
        Log in to ReactPop!
      </h1>
      <form
        class="loginForm"
      >
        <div
          class="formField formField--focused: loginForm-field"
        >
          <label
            class="labelform"
          >
            <span>
              Email
            </span>
            <input
              autocomplete="off"
              class="formField-input"
              name="email"
              type="email"
              value=""
            />
          </label>
        </div>
        <div
          class="formField loginForm-field"
        >
          <label
            class="labelform"
          >
            <span>
              Password
            </span>
            <input
              autocomplete="off"
              class="formField-input"
              name="password"
              type="password"
              value=""
            />
          </label>
        </div>
        <div
          class="formField loginForm-field"
        >
          <label
            class="labelform"
          >
            <span>
              Remember me
            </span>
            <input
              autocomplete="off"
              class="formField-input"
              name="remember"
              type="checkbox"
              value="false"
            />
          </label>
        </div>
        <button
          class="sc-bdvvtL iLUAWJ loginForm-submit"
          type="submit"
        >
          Log in!!
        </button>
      </form>
    </div>
  </div>
</div>
`;
